File: src/FactoryMethod/src/Temporada.java
import java.util.ArrayList;

public class Temporada {
    private int numero;
    private ArrayList<Episodio> episodios;

    public Temporada(int numero) {
        this.numero = numero;
        this.episodios = new ArrayList<>();
    }

    // Métodos getter e setter
    public int getNumero() {
        return numero;
    }

    public void setNumero(int numero) {
        this.numero = numero;
    }

    public ArrayList<Episodio> getEpisodios() {
        return episodios;
    }

    public void setEpisodios(ArrayList<Episodio> episodios) {
        this.episodios = episodios;
    }

    // Método para listar episódios
    public ArrayList<Episodio> listarEpisodios() {
        return episodios;
    }

    // Método para adicionar um episódio
    public void adicionarEpisodio(Episodio episodio) {
        this.episodios.add(episodio);
    }

    // Método para imprimir informações da temporada
    public void printarInformacoes() {
        System.out.println("Temporada " + numero);
        for (Episodio episodio : episodios) {
            episodio.printarInformacoes();
        }
    }
}

File: src/FactoryMethod/src/models/temporada.ts
import { Episodio } from './episodio';

export class Temporada {
    private numero: number;
    private episodios: Episodio[];

    constructor(numero: number) {
        this.numero = numero;
        this.episodios = [];
    }

    // Métodos getter e setter
    public getNumero(): number {
        return this.numero;
    }

    public setNumero(numero: number): void {
        this.numero = numero;
    }

    public getEpisodios(): Episodio[] {
        return this.episodios;
    }

    public setEpisodios(episodios: Episodio[]): void {
        this.episodios = episodios;
    }

    // Método para listar episódios
    public listarEpisodios(): Episodio[] {
        return this.episodios;
    }

    // Método para adicionar um episódio
    public adicionarEpisodio(episodio: Episodio): void {
        this.episodios.push(episodio);
    }

    // Método para imprimir informações da temporada
    public printarInformacoes(): void {
        console.log(`---- Temporada ${this.numero} ----`);
        for (const episodio of this.episodios) {
            episodio.printarInformacoes();
        }
    }
}